lib/os/sem.c

```diff
@@ -123,7 +123,7 @@ int sys_sem_give(struct sys_sem *sem)
 
 int sys_sem_take(struct sys_sem *sem, k_timeout_t timeout)
 {
-	int ret_value = 0;
+	int ret_value = k_sem_take(&sem->kernel_sem, timeout);
```

Comment:
It looks like k_sem_take() is getting called twice in a row.

Reply:
You are right, I fixed it.